pyamg/amg_core/ruge_stuben.h

```diff
@@ -956,345 +943,1339 @@ void cr_helper(const I A_rowptr[],  const int A_rowptr_size,
+                                T inner_denominator = 0;
+                                for (I ll = Sp[i]; ll < Sp[i+1]; ll++) {
+                                    I this_point = Sj[ll];
+
+                                    // Strong C-connections
+                                    if (splitting[this_point] == C_NODE) {
+                                        
+                                        // Add connection a_kl if present in matrix (search over kth row in A for connection)
+                                        // Only add if sign of a_kl does not equal sign of a_kk
+                                        for (I search_ind = Ap[k]; search_ind < Ap[k+1]; search_ind++) {
+                                            if (Aj[search_ind] == this_point) {
+                                                T a_kl = Ax[search_ind];
+                                                if (signof(a_kl) != signof(a_kk)) {
+                                                    inner_denominator += a_kl;
+                                                }
+                                                break;
+                                            }
+                                        }
+                                    }
+                                    // Strong F-connections (excluding self)
+                                    else if (this_point != i) {
+                                        for (I ff = Sp[this_point]; ff < Sp[this_point+1]; ff++) {
+                                            I d2_point = Sj[ff];
+
+                                            // Strong C-connections to strong F-connections (distance two C connections)
+                                            if (splitting[d2_point] == C_NODE) {
+
+                                                // Add connection a_kl if present in matrix (search over kth row in A
+                                                // for connection). Only add if sign of a_kl does not equal sign of a_kk
+                                                for (I search_ind = Ap[k]; search_ind < Ap[k+1]; search_ind++) {
+                                                    if (Aj[search_ind] == d2_point) {
+                                                        T a_kl = Ax[search_ind];
+                                                        if (signof(a_kl) != signof(a_kk)) {
+                                                            inner_denominator += a_kl;
+                                                        }
+                                                        break;
+                                                    }
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+
+                                // Add a_ki to inner denominator
```

Comment:
if combining ext and ext+i routines, this block of code (up to and including `inner_denominator += a_ki`) should be inside an `if(plusi)` statement